src/evaluator.rs

```diff
@@ -37,8 +37,22 @@ impl Environment {
 	}
 
 	pub fn add_binding(&mut self, name: String, val: LimpValue){
-		// TODO: lol this doesn't work
-		//self.bindings.last().unwrap().insert(name, val);
+		let len = self.bindings.len();
+
+		self.bindings[len-1].insert(name, val);
+	}
+
+	pub fn get_binding(&mut self, name: String) -> LimpValue {
+		let len = self.bindings.len();
+
+		for i in len..0 {
+			if self.bindings[i].contains_key(&name) {
+				let value = self.bindings[i].get(&name).unwrap();
+				return value.clone();
+			}
+		}
+
+		return ErrorValue;
```

Comment:
you can just replace this line with `ErrorValue` btw; if there is an expression instead of a statement, then Rust returns the expression. Seems a bit weird at first but it's idiomatic.

Reply:
ok, I'll edit it.